src/Traits/Statable.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace Sebdesign\SM\Traits;
+
+use SM\Factory\FactoryInterface;
+use SM\StateMachine\StateMachine;
+use Sebdesign\SM\Models\StateHistory;
+
+/**
+ * Trait Statable.
+ */
+trait Statable
+{
+    /**
+     * @var StateMachine
+     */
+    protected $stateMachine;
+
+    /**
+     * @return \Illuminate\Database\Eloquent\Model
+     */
+    public function history()
+    {
+        return $this->morphMany(StateHistory::class, 'statable');
+    }
+
+    /**
+     * @param array $transitionData
+     * @return $this|\Illuminate\Database\Eloquent\Model
+     */
+    public function addHistoryLine(array $transitionData)
+    {
+        $transitionData['actor_id'] = $this->getActorId();
+
+        return $this->history()->create($transitionData);
+    }
+
+    /**
+     * @return int|null
+     */
+    public function getActorId()
+    {
+        return auth()->id();
+    }
+
+    /**
+     * @return mixed|string
+     * @throws \Illuminate\Container\EntryNotFoundException
+     */
+    public function stateIs()
```

Comment:
This method should be named `getState()`.

Reply:
This causes a problem when someone uses `state` property of the model to store state. It looks like neither `state()` nor `getState()` works as it should in this case.

Reply:
I can suggest `currentState`, but I don't like it too much. I choose `stateIs` because it reads well, like
```
$article->stateIs() // new
```

Reply:
I just don't know why it is not working with `getState` then. The test just stops at the first method and nothing happens (using debug and verbose). I changed the property to something else and it just worked.

Reply:
This is the output:
```
phpunit --debug -c phpunit.xml.dist tests/Statable/StatableTest.php                                                                    [0fefc9b]
PHPUnit 6.2.3 by Sebastian Bergmann and contributors.

Runtime:       PHP 7.0.22-0ubuntu0.17.04.1
Configuration: /home/iben/Data/dev/laravel-state-machine/phpunit.xml.dist
Error:         No code coverage driver is available


Starting test 'Sebdesign\SM\Test\Statable\StatableTest::it_initiates_the_state_machine'.

```
It never exits.

Reply:
I'm not sure, but I think I have a clue why it fails: I found that SM uses `Symfony\Component\PropertyAccessor` to get the state of the object according to `property_path`. In its interface I found this at the `getValue` method (vendor/symfony/property-access/PropertyAccessorInterface.php:64):
>     * This method first tries to find a public getter for each property in the
>     * path. The name of the getter must be the camel-cased property name
>     * prefixed with "get", "is", or "has".
I think this will try to call `getState` on the object when instantiating the SM, which hits the trait method and tries to instantiate the SM again, and so on, infinite loop.

My only idea is `getCurrentState`, despite it's overly verbose. Any other suggestions?

Reply:
Oh I never thought of that. Another suggestion would be not to include a getter at all. The user probably could directly get the state attribute from the model if they know its `property_path` like `$model->state`, otherwise they could use `$model->stateMachine()->getState()`, which is verbose but convenient because there are no side effects.

Even I we used `getCurrentState` that could be a problem if the `property_path` was `current_state`, right?

Reply:
Right! Let's leave out the getter.